src/services/Auth.js

```diff
@@ -0,0 +1,43 @@
+import React, { useEffect, useState, createContext} from 'react';
+import { supabase } from './supabaseClient';
+
+export const AuthContext = createContext();
+
+export const  AuthProvider = ({ children }) => {
+
+    const [ user, setUser ] = useState();
+    const [ loading, setLoading ] = useState ('true');
+
+    useEffect(()=>{
+        const session = supabase.auth.session();
+
+        setUser (session ?. user ?? null );
+        setLoading('false')
```

Comment:
never write booleans (`true` / `false`) as strings (`'true'`).

`'false'` will equal `true`!

Reply:
Thanks I changed the booleans. If it id possible, I don't want to add signing and signet for now.
 Because of the error I couldn't replace the supbase.user in routes or other place, but I'm going to replace it.